sdk/schemaregistry/azure-schemaregistry-avroencoder/azure/schemaregistry/encoder/avroencoder/_schema_registry_avro_encoder.py

```diff
@@ -205,47 +251,54 @@ def encode(
 
         if message_type:
             try:
-                return message_type.from_message_content(payload, content_type, **kwargs)
-            except AttributeError:
-                try:
-                    return message_type(payload, content_type, **kwargs)
-                except TypeError as e:
-                    SchemaEncodeError(
-                        f"""The content model {str(message_type)} is not a Callable that takes `content`
-                            and `content_type` or a subtype of the MessageType protocol.
-                            If using an Azure SDK model class, please check the README.md for the full list
-                            of supported Azure SDK models and their corresponding versions."""
-                    ).raise_with_traceback()
+                return message_type.from_message_content(
+                    payload, content_type, **kwargs
+                )
+            except AttributeError as e:
+                raise AvroEncodeError(
```

Comment:
I think this should be a TypeError, but still wanted a way to return the content/content_type values in the exception. Since the content has already been encoded, user can recover from this scenario. However, didn't think we would create a new error just for this.

Thoughts on alternatives?

Reply:
raise as TypeError and pass message_content dict as an arg.